uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt311/client/paho/Mqtt311ConnectionImpl.java

```diff
@@ -280,4 +267,59 @@ private void checkCorrelationData(byte[] correlationData) {
             logger.atWarn().log("MQTT v3.1.1 doesn't support correlation data");
         }
     }
+
+    class MqttCallbackImpl implements MqttCallback {
+
+        @SuppressWarnings("PMD.AvoidCatchingGenericException")
+        @Override
+        public void connectionLost(Throwable throwable) {
+            // only unsolicited disconnect
+            if (isClosing.get()) {
+                logger.atWarn().log("DISCONNECT event ignored due to shutdown initiated");
+            } else {
+                GRPCClient.DisconnectInfo disconnectInfo = new GRPCClient.DisconnectInfo(null, null, null, null, null);
+                executorService.submit(() -> {
+                    try {
+                        grpcClient.onMqttDisconnect(connectionId, disconnectInfo, throwable.getMessage());
+                    } catch (Exception ex) {
+                        logger.atError().withThrowable(ex).log("onMqttDisconnect failed");
+                    }
+                });
+            }
+
+            logger.atInfo().log("MQTT connection {} interrupted, error code {}", connectionId, throwable.getMessage());
+
+        }
+
+        @Override
+        public void messageArrived(String topic, MqttMessage mqttMessage) {
+            Mqtt311ConnectionImpl.this.messageArrived(topic, mqttMessage);
+        }
+
+        @Override
+        public void deliveryComplete(IMqttDeliveryToken token) {
+            logger.atInfo().log("Delivery completion is {}", token.isComplete());
+        }
+    }
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private void messageArrived(String topic, MqttMessage mqttMessage) {
```

Comment:
Just rename private method to avoid infinite recusion by mistake.

Reply:
fixed in dfd2cb402305dcd47a03070e302a2a1f0592d8f7